### PR TITLE
Get block certificates

### PR DESCRIPTION
**Pull request by magnusbechwind**
## Purpose

Support the new `GetBlockCertificates` query.
## Changes

Added classes: `BlockCertificates`, `EpochFinalizationEntry`, `FinalizerRound`, `QuorumCertificate`, `QuorumSignature`, `SuccessorProof`, `TimeoutCertificate` and `TimeoutSignature`.
Added tests and an example of the implemented method.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2.java [hunk 1]
```diff
@@ -775,6 +776,18 @@ public InvokeInstanceResult invokeInstance(InvokeInstanceRequest request) {
         return InvokeInstanceResult.parse(grpcResponse);
     }
 
+    /**
+     * Retrieves the {@link BlockCertificates} for a given block.
+     * Note that, if the block being pointed to is not a product of ConcordiumBFT, then a INVALID_ARGUMENT exception will be thrown.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/ClientV2.java:781]: Maybe point out that this is the case for blocks that was created before protocol version 6.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/blockcertificates/EpochFinalizationEntry.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * The epoch finalization entry is the proof that makes the protocol able to advance to a new epoch.
+ * I.e. the {@link EpochFinalizationEntry} is present if and only if the block is the first block of a new {@link Epoch}.
+ */
+@Builder
+@Getter
+@ToString
+@EqualsAndHashCode
+
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/blockcertificates/EpochFinalizationEntry.java:17]: I don't think this newline is necessary :)

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/blockcertificates/QuorumSignature.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+@Builder
+@Getter
+@EqualsAndHashCode
+@ToString
+public class QuorumSignature {
+
+    /**
+     * The bytes representing the raw aggregate signature. The bytes have a fixed length of 48 bytes.
+     */
+    private final byte[] value;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/blockcertificates/QuorumSignature.java:20]: Don't we have a BLSSignature class?

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/blockcertificates/SuccessorProof.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ */
+@Getter
+@Builder
+@EqualsAndHashCode
+@ToString
+public class SuccessorProof {
+    /**
+     * The proof represented as raw bytes. The bytes have a fixed length of 32 bytes.
+     */
+    private final byte[] value;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/blockcertificates/SuccessorProof.java:21]: This is a sha256 hash.

**Review by EmilLa1 — COMMENTED** [review]
Looks good! 
Minor comments regarding documentation where I think we can improve a bit.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## Unreleased changes
+- Added support for GRPC V2 `GetBlockCertificates` for retrieving certificates for a block supporting ConcordiumBFT
```
Comment (EmilLa1) [inline CHANGELOG.md:4]: Please write that this feature requires a node with version at least 6.1

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/blockcertificates/QuorumCertificate.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import com.concordium.sdk.responses.Round;
+import com.concordium.sdk.transactions.Hash;
+import lombok.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A quorum certificate is the certificate that the finalization committee issues in order to certify a block.
+ * A block must be certified before it will be part of the authoritative part of the chain.
```
[the remark this answers is not in the document]
Comment (magnusbechwind) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/blockcertificates/QuorumCertificate.java:14]: I added this clarification to the `BlockCertificates` class as it seems most applicable there?

**Review by EmilLa1 — APPROVED** [review]
looks good!

**Pull request merged**